backend/apps/mentorship/api/internal/mutations/module.py

```
@@ -342,8 +342,28 @@ def update_module(self, info: strawberry.Info, input_data: UpdateModuleInput) ->
-            raise PermissionDenied
+        # Check if user is a program admin OR a mentor assigned to this module
+        is_program_admin = module.program.admins.filter(id=creator_as_mentor.id).exists()
+        is_module_mentor = module.mentors.filter(id=creator_as_mentor.id).exists()
+
+        if not (is_program_admin or is_module_mentor):
+            msg = "You do not have permission to edit this module. Only program admins and assigned mentors can edit modules."
+            logger.warning(
+                "User '%s' attempted to edit module '%s' without proper permissions.",
+                user.username,
+                input_data.key,
+                exc_info=True,
```

Ruff (0.14.10)

`exc_info=` outside exception handlers

Remove `exc_info=`

(LOG014)